File: youlexuanparent/youlexuan-sellergoods-interface/src/main/java/com/sunxupeng/sellergoods/service/ItemCatService.java
package com.sunxupeng.sellergoods.service;

import com.sunxupeng.entity.PageResult;
import com.sunxupeng.pojo.TbItemCat;

import java.util.List;

/**
 * item_cat服务层接口
 * @author sunxupeng
 *
 */
public interface ItemCatService {

	/**
	 * 返回全部列表
	 * @return
	 */
	public List<TbItemCat> findAll();
	
	
	/**
	 * 返回分页列表
	 * @return
	 */
	public PageResult findPage(int pageNum, int pageSize);


	/**
	 * 增加
	*/
	public void add(TbItemCat itemCat);


	/**
	 * 修改
	 */
	public void update(TbItemCat itemCat);


	/**
	 * 根据ID获取实体
	 * @param id
	 * @return
	 */
	public TbItemCat findOne(Long id);


	/**
	 * 批量删除
	 * @param ids
	 */
	public String delete(Long[] ids);

	/**
	 * 分页
	 * @param pageNum 当前页 码
	 * @param pageSize 每页记录数
	 * @return
	 */
	public PageResult findPage(TbItemCat itemCat, int pageNum, int pageSize);


	/**
	 * 根据上级ID返回列表
	 * @return
	 */
	public List<TbItemCat> findByParentId(Long parentId);
}

File: youlexuanparent/youlexuan-shop-web/src/main/webapp/js/service/itemCatService.js
//item_cat服务层
app.service('itemCatService', function($http){
	// 保存、修改
	this.save = function(entity, parentId) {
		var methodName = 'add'; 	// 方法名称
		if (entity.id != null) { 	// 如果有ID
			methodName = 'update'; 	// 则执行修改方法
		} else {
			entity.parentId = parentId;
		}

		return $http.post('../itemCat/' + methodName + '.do', entity);
	}

	// 查询单个实体
	this.findOne = function(id) {
		return $http.get('../itemCat/findOne.do?id=' + id);
	}
//查询全部用于显示分类名
	this.findAll = function() {
		return $http.get('../itemCat/findAll.do');
	}

	// 批量删除
	this.dele = function(ids) {
		// 获取选中的复选框
		return $http.get('../itemCat/delete.do?ids=' + ids);
	}

	// 查询
	this.search = function(page, size, searchEntity) {
		// post提交，page、size属性和之前相同，将searchEntity提交至后台@RequestBody对应的属性
		return $http.post('../itemCat/search.do?page=' + page + '&size=' + size,
				searchEntity);
	}
	//下拉列表数据
	this.selectOptionList=function(){
		return $http.get('../typeTemplate/selectOptionList.do');
	}
	this.findByParentId=function(parentId){
		return $http.get('../itemCat/findByParentId.do?parentId='+parentId);
	}

	this.findAll = function() {
		return $http.get('../itemCat/findAll.do');
	}
});